tools/azsdk-cli/docs/specs/99-repo-tooling-contract.md

```diff
@@ -0,0 +1,322 @@
+
+Managing parameter mappings can still cause tight coupling or potential breakages.
+The design proposal below outlines the use of `[Alias]` for powershell cmdlets to address this.
+However, this approach means that direct calls to repository tools (e.g. `msbuild`) should be made from the CLI, or in
+some special cases wrapped in a powershell script.
+
+Adding schemas for parameters could improve reliability but at the cost of requiring changes to `azsdk` and/or scripts to be coordinated.
+It would also require additional up front work to capture parameter schemas for existing scripts, along with CI checks to ensure the
+scripts implement the schema.
+
+## Design Proposal
+
+The original idea for this approach is copied from the [package updater spec](https://github.com/Azure/azure-sdk-tools/blob/cc5bae3dd44b02c01ebb8d7546fdbeb0838b665d/tools/azsdk-cli/docs/specs/6-package-updater.spec.md#proposed-approaches-for-the-intergration-between-language-specific-logic-and-tools) by [@raych1](http://github.com/raych1):
+
+> Define a small, well-documented contract/API (parameters, expected inputs/outputs, exit codes, JSON schema). Language repo implements the contract (script or program). CLI does normalization, validation, and invokes the repo-side contract. Optionally implement language-specific logic in CLI for cases where shared helpers or BCL types are beneficial; otherwise call the repo script.
+
+> Pros: Best of both worlds: a small, stable contract (parameters, exit codes, JSON output) that CLI depends on while letting repo implement details. Repo can evolve implementation without breaking consumers so long as contract remains stable. CLI can provide shared normalization, validation, telemetry, and orchestration, while repo scripts handle repo-local tasks (file paths, tool variants). Easier to test and mock (CLI tests against a known contract; repo tests ensure script adheres to contract).
+
+> Cons: Requires upfront design of the contract and discipline to keep it stable. Still requires versioning decisions (how to roll out contract changes across repos). Some duplication of wrapper code may persist (but minimized if contract is well-designed).
+
+The design proposal below removes some aspects of the contract initially proposed, only keeping the script path and lookup tags.
+
+### Overview
+
+When a command/tool is invoked in `azsdk` that is known to be repository-specific (i.e. the command requires a package path or a call to get the repository root), it should first check a centralized config in that repository. If the config contains an override entry for the command, call the override specified. If not, fall back to the implementation in the CLI/MCP server (or fail/no-op if not implemented).
+
+### Detailed Design
+
+In `eng/azsdk-cli-command-overrides.json` in a language repository, a set of mappings can be defined that
+correspond to CLI commands and/or MCP tools.
+
+```
+[
+  {
+    "tags": ["FooBarCheck", "azsdk_foo_bar", "foobar"],
```

Comment:
I understand "FooBarCheck" is implementation within `azsdk`, "foobar" is command part of `azsdk foobar ...` then what does "azsdk_foo_bar" mean? Should language owner add all these 3 name into `tag` when trying to override to use script of language repo?

Reply:
It's added in [this PR](https://github.com/Azure/azure-sdk-tools/pull/12714/files#diff-18ce93b1228809d2436033bc7e9b2dc3cd8df6a1d0e3aad8ff4d044e579149ddR144-R165)